internal/rbd/diskusage.go

```diff
@@ -23,7 +23,18 @@ import (
 // Sparsify checks the size of the objects in the RBD image and calls
 // rbd_sparify() to free zero-filled blocks and reduce the storage consumption
 // of the image.
+// This function will return ErrImageInUse if the image is in use, since
+// sparsifying an image on which i/o is in progress is not optimal.
```

Comment:
I don't want to sound like a broken record, but I have to ask again: do you have any data that would support automating `rbd sparsify` runs at all?  In particular:
- Have you done any research into how likely they are to be encountered?
- Do you have `ceph df` outputs captured before and after `rbd sparsify` is run for typical customer scenarios?

If the answer is no, I don't understand why `rbd sparsify` is being conditioned on `isInUse()` instead of just being dropped entirely (to be resurrected as an optional step when the relevant CSI addon spec allows taking options from the user).   Sparsifying the image the way Ceph CSI does it is likely not optimal in general, not just when there is competing I/O.

Reply:
on one heavily used cluster, I suspect I've seen 33% to 100% extra usage without sparsify. Hadn't had the chance to run it offline and haven't had the  guts to run it online. This makes it sound like it was good for me to be hesitent.

Reply:
Yeah -- I would suggest starting with `fstrim` (assuming you aren't using raw block PVs).